Sources/Authgear.swift

```diff
@@ -480,17 +481,18 @@ public class Authgear {
 
             let userInfo = try apiClient.syncRequestOIDCUserInfo(accessToken: oidcTokenResponse.accessToken!)
 
-            let result = persistSession(oidcTokenResponse, reason: .authenticated)
-                .flatMap {
+            let handler: VoidCompletionHandler = { result in
+                let newResult = result.flatMap {
                     Result { () in
                         if #available(iOS 11.3, *) {
                             try self.disableBiometric()
                         }
                     }
                 }
                 .map { userInfo }
-            return handler(result)
-
+                return handler(newResult)
+            }
+            persistSession(oidcTokenResponse, reason: .authenticated, handler: handler)
```

Comment:
It's more idiomatic to provide completion handler like this:
```swift
persistSession(oidcTokenResponse, reason: .authenticated) { result in
  handler(result.flatMap {
    Result { () in
      if #available(iOS 11.3, *) {
        try self.disableBiometric()
      }
    }
  })
}
```

Please check if you can simplify other changes like this.